tests/suite/test_transport_server_tcp_load_balance.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+import re
+from settings import TEST_DATA
+from suite.resources_utils import get_ingress_nginx_template_ts_conf
+from suite.resources_utils import (
+    wait_before_test,
+    patch_deployment_from_yaml
+)
+
+
+@pytest.mark.ts
+@pytest.mark.parametrize(
+    "crd_ingress_controller, transport_server_setup",
+    [
+        (
+            {
+                "type": "complete",
+                "extra_args":
+                    [
+                        "-enable-custom-resources",
+                        "-global-configuration=nginx-ingress/nginx-configuration",
+                        "-enable-leader-election=false"
+                    ]
+            },
+            {"example": "transport-server-status", "app_type": "simple"},
+        )
+    ],
+    indirect=True,
+)
+class TestTransportServerTcpLoadBalance:
+
+    def restore_service(self, kube_apis, transport_server_setup) -> None:
+        """
+        Function to revert a TransportServer resource to a valid state.
+        """
+        dns_file = f"{TEST_DATA}/transport-server-tcp-load-balance/standard/dns.yaml"
+        patch_deployment_from_yaml(kube_apis.apps_v1_api, transport_server_setup.namespace, dns_file)
+
+    @pytest.mark.sean
+    def test_number_of_replicas(
+        self, kube_apis, crd_ingress_controller, transport_server_setup, ingress_controller_prerequisites
+    ):
+        """
+        The load balancing of TCP should result in 3 servers to match the 3 replicas of a service.
+        """
+        dns_file = f"{TEST_DATA}/transport-server-tcp-load-balance/more-replicas.yaml"
+        patch_deployment_from_yaml(kube_apis.apps_v1_api, transport_server_setup.namespace, dns_file)
```

Comment:
To control the number of replicas, another approach is to only to update the scale subresource using  `apps_v1_api.patch_namespaced_deployment_scale()`  ( which corresponds to  Patch Scale in https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#deployment-v1-apps )
This way, there will be no need to have multiple YAMLS for different number of replicas - it will be possible just to patch the number of replicas
This would be similar to the following kubectl command:
```
kubectl scale deploy coffee --replicas=1 -v=8
. . .
I0419 16:01:17.808008   45385 request.go:968] Request Body: {"spec":{"replicas":1}}
. . .

Reply:
done